crates/typst-syntax/src/parser.rs

```diff
@@ -633,6 +633,9 @@ fn code_exprs(p: &mut Parser, mut stop: impl FnMut(&Parser) -> bool) {
             code_expr(p);
             if !p.end() && !stop(p) && !p.eat_if(SyntaxKind::Semicolon) {
                 p.expected("semicolon or line break");
+                if p.at(SyntaxKind::Label) {
+                    p.hint("you can use labels only in markup mode");
```

Comment:
Maybe we can improve this message a bit? Something like "labels can only be applied in markup mode", followed by a second hint "try wrapping your code in a markup block (\`[ ]\`)"

(I'm open to further suggestions)